### PR TITLE
feat: allow pulling queries from `system.profile`

### PR DESCRIPTION
**Pull request by jsalvata**
#### Changes Made

* Created a 2nd method capable of pulling problematic queries from `system.profile` (instead of just from currently running queries)
* Added option to use that method instead of the original one
* Added option to add queries requiring a COLLSCAN to the results
* Added a flag to the result objects indicating if the queries require a COLSCAN

#### Potential Risks
I have not completely checked the accuracy of the output (which relies on the exact semantics of the `system.profile.planSummary` property. There may be operations requiring COLLSCAN in sub-queries (e.g. aggregations) going unnoticed.

#### Test Plan
- [x] Start the mongo server with `--profile 2`. Run the service and check the output.

#### Checklist
- [ ] I've increased test coverage
- [x] Since this is a public repository, I've checked I'm not publishing private data in the code, commit comments, or this PR.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -57,6 +66,34 @@ class MongoSlowQueryChecker {
...
+  }
+
+  /**
+   * Get relevant (slow/COLLSCAN) queries from the `system.profile` table, ensuring we don't
+   * repeatedly return the same queries in consecutive runs.
+   */
+  async getFromSystemProfile() {
+    this.profile = this.profile || (await this.db.connect()).collection('system.profile');
+
+    const conditions = [{ millis: { $gte: this.queryThreshold * 1000 } }];
```
Comment (shils) [inline src/index.js:88]: We should note that the `system.profile` approach only returns queries longer than both `this.queryThreshold`  and the mongod servers slow query threshold.

File: src/index.js [hunk 2]
```diff
@@ -57,6 +66,34 @@ class MongoSlowQueryChecker {
...
+    if (this.reportAllCollscans) conditions.push({ planSummary: { $regex: /COLLSCAN/ } });
```
Comment (shils) [inline src/index.js:89]: Are `COLLSCAN`s included in `system.profile` even if they don't exceed the servers slow query threshold?

Comment (jsalvata) [reply src/index.js:89]: No. You have to use a zero threshold to get this. I'll document this requirement.

File: src/index.js (hunk 1 above)
Comment (jsalvata) [reply src/index.js:88]: Done.

File: src/index.js (hunk 2 above)
Comment (jsalvata) [reply src/index.js:89]: Done.

**Review by shils — APPROVED** [review]

**Pull request merged**